s3transfer/futures.py

```diff
@@ -422,38 +536,91 @@ def shutdown(self, wait=True):
 
 
 class ExecutorFuture(object):
-    def __init__(self, future):
-        """A future returned from the executor
+    def __init__(self):
+        """A future returned from the executor.
 
-        Currently, it is just a wrapper around a concurrent.futures.Future.
-        However, this can eventually grow to implement the needed functionality
-        of concurrent.futures.Future if we move off of the library and not
-        affect the rest of the codebase.
-
-        :type future: concurrent.futures.Future
-        :param future: The underlying future
+        This is a pared down version of concurrent.futures.Future, stripping
+        away all the features that we don't use and modifying the interface
+        somewhat (namely removing the argument from callbacks).
         """
-        self._future = future
+        self._condition = threading.Condition()
+        self._done = False
+        self._result = None
+        self._exception = None
+        self._traceback = None
+        self._done_callbacks = []
 
-    def result(self):
-        return self._future.result()
+    def _invoke_callbacks(self):
+        """Invokes all the queued done callbacks."""
+        for callback in self._done_callbacks:
+            try:
+                callback()
+            except Exception:
+                logger.debug('exception calling callback for %r', self)
```

Comment:
I realize that this is from the futures source code, but I wonder if it is worth capturing the traceback as well in the log.